python_modules/dagster/dagster/_core/execution/context/invocation.py

```diff
@@ -375,186 +416,8 @@ def get_output_metadata(
     def get_mapping_key(self) -> Optional[str]:
         return self._mapping_key
 
-
-def _validate_resource_requirements(
-    resource_defs: Mapping[str, ResourceDefinition], op_def: OpDefinition
-) -> None:
-    """Validate correctness of resources against required resource keys."""
-    if cast(DecoratedOpFunction, op_def.compute_fn).has_context_arg():
-        for requirement in op_def.get_resource_requirements():
-            if not requirement.is_io_manager_requirement:
-                ensure_requirements_satisfied(resource_defs, [requirement])
-
-
-class BoundOpExecutionContext(OpExecutionContext):
-    """The op execution context that is passed to the compute function during invocation.
-
-    This context is bound to a specific op definition, for which the resources and config have
-    been validated.
-    """
-
-    _op_def: OpDefinition
```

Comment:
I don't really understand what these were in the BoundOpExecutionContext for. I can add them to the new class though if they are needed!

Reply:
For historical context here - `op_def` was added so that after you've actually invoked an op, you could access definition-level properties about that op on the context object itself. This is useful for the situation where you use op factories for example, and might still need to query definition-level information.

Reply:
I thought since these were on the `BoundExecutionContext` that they wouldn't be accessible after invocation since the bound context gets garbage collected

Reply:
right - i meant during invocation.

Reply:
ah ok - this should still work in the new setup